### PR TITLE
changes for initialiseAsync to be parallel

### PR DESCRIPTION
**Pull request by nasimafeef**

### CODE REVIEW

**Review by Javasamurai — APPROVED** [review]

**Comment by rampr** [comment]
can't we move this to index.html? We have to add it to the build file. At some point we need to report loading progress. It will also help from that perspective.

But even without the loading progress, moving it to index.html should help

File: build/index.js [hunk 1]
```diff
@@ -2,7 +2,16 @@ exports.onBeforeBuild = function (api, app, config, cb) {
   if (config.browser) {
     if (config.browser.headHTML) {
       config.browser.headHTML.push(
-        '<script src="https://connect.facebook.net/en_US/fbinstant.6.2.js"></script>'
+        '<script src="https://connect.facebook.net/en_US/fbinstant.6.2.js"></script> \
+        <script> \
+          if (window.FBInstant) { \
+            window.FBInstant.initializeAsync() \
+            .then(function () { \
```
Comment (rampr) [inline build/index.js]: minor indentation issue

Comment (rampr) [reply build/index.js]: actually ignore this, it is a string

**Review by rampr — APPROVED** [review]

File: build/index.js (hunk 1 above)
Comment (nasimafeef) [reply build/index.js]: i corrected it anyway.

File: js/index.js [hunk 2]
```diff
@@ -38,11 +22,14 @@ var FacebookInstant = Class(function () {
...
   };
 
-  this.setPaymentsReady = function () {
+  this.setPaymentsReady = function (reg_cb) {
     this.onReady(bind(this, function () {
       if (this.FBInstant.getSupportedAPIs()
         .includes('payments.purchaseAsync')) {
         this.payments_ready = true;
+        if (reg_cb) {
+          reg_cb.fn(reg_cb.products, reg_cb.next);
```
Comment (vivekb91) [inline js/index.js:34]: Add a TODO here

**Review by vivekb91 — APPROVED** [review]

**Review by Javasamurai — APPROVED** [review]

**Pull request merged**